python/main.py

```diff
@@ -98,8 +112,78 @@ async def get_image(image_name):
 class Item(BaseModel):
     name: str
     category: str
+    image_name: str
 
+DEFAULT_JSON_DATA = {"items": []}
 
 def insert_item(item: Item):
     # STEP 4-1: add an implementation to store an item
-    pass
+    try:
+        if not JSON_DB.exists():
+            with open(JSON_DB, "w", encoding="utf-8") as f:
+                json.dump(DEFAULT_JSON_DATA, f, indent=2)
+
+        with open(JSON_DB, "r+", encoding="utf-8") as f:
+            content = f.read().strip()
+            data = json.loads(content) if content else DEFAULT_JSON_DATA
+            logger.info("Succeeded to open json file")
+
+            if "items" not in data:
+                data["items"] = []
+
+            existing_item = next((i for i in data["items"] if i["name"] == item.name), None)
+
+            if existing_item:
+                logger.info(f"Item already exists, updating image_name: {existing_item}")
+            else:
+                new_item = item.dict()
+                data["items"].append(new_item)
+                logger.info(f"New item inserted: {new_item}")
+
+            f.seek(0)
+            json.dump(data, f, indent=2, ensure_ascii=False)
+            f.truncate()
+
+    except Exception as e:
+        logger.error(f"Failed to save item: {e}")
+        raise HTTPException(status_code=500, detail="Failed to save item")
+
+
+
+@app.get("/items")
+def get_items():
+    try:
+        if not JSON_DB.exists():
+            return {"items": []}
+
+        with open(JSON_DB, "r", encoding="utf-8") as f:
+            content = f.read().strip()
+            data = json.loads(content) if content else {"items": []}
```

Comment:
We can probably use the `DEFAULT_JSON_DATA` here as well
```suggestion
            data = json.loads(content) if content else DEFAULT_JSON_DATA
```